frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx

```diff
@@ -223,13 +223,34 @@ export const AgentNotRunningButton = () => {
       }
     }
 
-    if (isNil(service) && isNil(middlewareServiceResponse))
-      throw new Error('Service not found');
+    // Update if service already exists and template hash has changed
+    if (!createServiceResponse && service) {
+      try {
+        if (service.hash !== serviceTemplate.hash) {
+          await ServicesService.updateService({
+            serviceConfigId: service.service_config_id,
+            stakingProgramId: selectedStakingProgramId,
+            chainId: selectedAgentConfig.evmHomeChainId,
+            serviceTemplate,
+            deploy: true,
+            useMechMarketplace:
+              STAKING_PROGRAMS[selectedAgentConfig.evmHomeChainId][
+                selectedStakingProgramId
+              ].mechType === MechType.Marketplace,
+          });
+        }
+      } catch (error) {
+        console.error('Error while updating the service:', error);
+        showNotification?.('Failed to update service.');
+        throw new Error('Failed to update service');
+      }
+    }
 
-    // Start the service
+    // Start service if hasn't just been created
     try {
-      const serviceToStart = service ?? middlewareServiceResponse;
-      await ServicesService.startService(serviceToStart!.service_config_id);
+      if (service && !createServiceResponse) {
```

Comment:
does create also run the service now? so we shouldn't start manually?

Reply:
good catch thanks, should be the other way round i think 😓